### PR TITLE
make `kubectl explan` test tolerate either  "RESOURCE" or "FIELD"

### LINKED ISSUE #116650: [Failing Test] kubeadm-kinder-upgrade-1-26-latest,kubeadm-kinder-latest
### Which jobs are failing?

master-informing

- kubeadm-kinder-upgrade-1-26-latest
- kubeadm-kinder-latest

### Which tests are failing?

`Kubernetes e2e suite: [It] [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for CRD with validation schema [Conformance]`

### Since when has it been failing?

03-15 04:46 IST

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#kubeadm-kinder-upgrade-1-26-latest

### Reason for failure (if possible)

```
{ failed [FAILED] kubectl explain e2e-test-crd-publish-openapi-6480-crds.spec.bars result {KIND:     e2e-test-crd-publish-openapi-6480-crd
VERSION:  crd-publish-openapi-test-foo.example.com/v1

RESOURCE: bars <[]Object>

DESCRIPTION:
     List of Bars and their specs.

FIELDS:
   age	<string>
     Age of Bar.

   bazs	<[]string>
     List of Bazs.

   feeling	<string>
     Whether Bar is feeling great.

   name	<string> -required-
     Name of Bar.

} doesn't match pattern {(?s)FIELD:.*bars.*<\[\]Object>.*DESCRIPTION:.*List of Bars and their specs.*FIELDS:.*bazs.*<\[\]string>.*List of Bazs.*name.*<string>.*Name of Bar}
In [It] at: test/e2e/apimachinery/crd_publish_openapi.go:133 @ 03/15/23 13:53:06.423
}
```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig api-machinery

### PR DESCRIPTION
**Pull request by pacoxu**
Fixes #116650

### CODE REVIEW

File: test/e2e/apimachinery/crd_publish_openapi.go [hunk 1]
```diff
@@ -129,12 +129,12 @@ var _ = SIGDescribe("CustomResourcePublishOpenAPI [Privileged:ClusterAdmin]", fu
 		if err := verifyKubectlExplain(f.Namespace.Name, crd.Crd.Spec.Names.Plural+".spec", `(?s)DESCRIPTION:.*Specification of Foo.*FIELDS:.*bars.*<\[\]Object>.*List of Bars and their specs`); err != nil {
 			framework.Failf("%v", err)
 		}
-		if err := verifyKubectlExplain(f.Namespace.Name, crd.Crd.Spec.Names.Plural+".spec.bars", `(?s)FIELD:.*bars.*<\[\]Object>.*DESCRIPTION:.*List of Bars and their specs.*FIELDS:.*bazs.*<\[\]string>.*List of Bazs.*name.*<string>.*Name of Bar`); err != nil {
```
Comment (lavalamp) [inline test/e2e/apimachinery/crd_publish_openapi.go:132]: I think we want to make it tolerate either string, not only the incorrect "RESOURCE"

File: test/e2e/apimachinery/crd_publish_openapi.go [hunk 2]
```diff
@@ -129,12 +129,12 @@ var _ = SIGDescribe("CustomResourcePublishOpenAPI [Privileged:ClusterAdmin]", fu
...
+		if err := verifyKubectlExplain(f.Namespace.Name, crd.Crd.Spec.Names.Plural+".spec.bars", `(?s)RESOURCE:.*bars.*<\[\]Object>.*DESCRIPTION:.*List of Bars and their specs.*FIELDS:.*bazs.*<\[\]string>.*List of Bazs.*name.*<string>.*Name of Bar`); err != nil {
 			framework.Failf("%v", err)
 		}
 
 		ginkgo.By("kubectl explain works to return error when explain is called on property that doesn't exist")
-		if _, err := e2ekubectl.RunKubectl(f.Namespace.Name, "explain", crd.Crd.Spec.Names.Plural+".spec.bars2"); err == nil || !strings.Contains(err.Error(), `field "bars2" does not exist`) {
+		if _, err := e2ekubectl.RunKubectl(f.Namespace.Name, "explain", "--output", "plaintext-openapiv2", crd.Crd.Spec.Names.Plural+".spec.bars2"); err == nil || !strings.Contains(err.Error(), `field "bars2" does not exist`) {
```
Comment (alexzielenski) [inline test/e2e/apimachinery/crd_publish_openapi.go:137]: Is using the old output format necessary here? I believe this part of the test works both for `explain` backends

File: test/e2e/apimachinery/crd_publish_openapi.go [hunk 3]
```diff
@@ -716,7 +716,7 @@ func dropDefaults(s *spec.Schema) {
 }
 
 func verifyKubectlExplain(ns, name, pattern string) error {
-	result, err := e2ekubectl.RunKubectl(ns, "explain", name)
+	result, err := e2ekubectl.RunKubectl(ns, "explain", "--output", "plaintext-openapiv2", name)
```
Comment (alexzielenski) [inline test/e2e/apimachinery/crd_publish_openapi.go:719]: same comment as the other one

File: test/e2e/apimachinery/crd_publish_openapi.go (hunk 1 above)
Comment (pacoxu) [reply test/e2e/apimachinery/crd_publish_openapi.go:132]: It seems the last fix is not correct. 
I updated the PR to tolerate either string FIELD or RESOURCE.

File: test/e2e/apimachinery/crd_publish_openapi.go (hunk 2 above)
Comment (pacoxu) [reply test/e2e/apimachinery/crd_publish_openapi.go]: Removed the change here.

**Comment by lavalamp** [comment]
/lgtm
/approve

Thanks!

**Comment by aojea** [comment]
/milestone v1.27

this is a release blocking problem

**Pull request merged**